src/Microsoft.AspNetCore.SignalR.Core/DefaultHubLifetimeManager.cs

```diff
@@ -83,7 +83,11 @@ private Task SendAllWhere(string methodName, object[] args, Func<HubConnectionCo
                     continue;
                 }
 
-                tasks.Add(connection.WriteAsync(message));
+                try
+                {
+                    tasks.Add(connection.WriteAsync(message));
+                }
+                catch { }
```

Comment:
Do we want to log?

Reply:
Isn't `WriteAsync` returning a `Task` and here you're simply adding a `Task` to a list. Does that need a try/catch? Isn't it the await that needs a try/catch?

Reply:
It looks like `WriteAsync` starts with and `await`, so it's not likely to be an issue in practice. This would be more of a defense-in-depth kind of thing (so that if it's refactored to do argument validation or serialization synchronously, it doesn't have a risk of killing the whole loop).

Perhaps we should be doing `tasks.Add(Task.FromException(ex))`, so that the caller still gets the error? Or maybe this is overkill.

Reply:
The `async` keyword on `WriteAsync` should generate a state machine that will always catch exceptions and return them as a faulted `Task` instead of throwing out of the method, at least that is my understanding of async/await.

Reply:
That's correct but this shouldn't start failing if we remove the state machine from WriteAsync (which we'll very likely do).

Reply:
When that happens shouldn't the try/catch should be inside WriteAsync and return and `Task.FromException(ex)`?

Reply:
That would make the calling code consistent

Reply:
> When that happens shouldn't the try/catch should be inside WriteAsync and return and Task.FromException(ex)?

Ugh, I dislike that but I see why it's an attractive option. try catch is fine..... 😄. I don't know if there's a best practice here tbh. For e.g. if you're doing argument validation, should it throw async or sync?

Reply:
Argument validation should throw sync. `JsonTextWriter.WritePropertyNameAsync(null)` for example will throw a `NullArgumentException` instead of returning a faulted task, but any internal errors like the underlying stream being disposed will return a faulted task.

Similar to `Task.Delay(-1000)` throwing an argument exception instead of returning a faulted task.

Reply:
Which is why I like having a try catch

Reply:
We control the method so we can choose not to do that.

What about a `SafeWriteAsync` helper method that wraps WriteAsync and has the try/catch? If we need it wrapped for safety then it would be nicer that it was in one place.

Reply:
That sounds good.